pallets/staking/src/lib.rs

```diff
@@ -2665,6 +2728,23 @@ decl_module! {
             SlashingAllowedFor::put(slashing_switch);
             Self::deposit_event(RawEvent::SlashingAllowedForChanged(slashing_switch));
         }
+
+        /// Scale the intended validator count for a given DID.
+        ///
+        /// #Arguments
+        /// * origin Required origin for adding a potential validator.
+        /// * identity Validator's IdentityId.
+        /// * additional Amount by which `intended_count` gets increased.
+        #[weight = 150_000_000]
+        pub fn scale_permissioned_validator_intended_count(origin, identity: IdentityId, additional: u32) {
+            T::RequiredAddOrigin::ensure_origin(origin)?;
+            ensure!(Self::permissioned_identity(&identity).is_some(), Error::<T>::NotExists);
+            PermissionedIdentity::mutate(&identity, |pref| {
+                if let Some(p) = pref {
+                    p.intended_count += additional
+                }
+            });
```

Comment:
```suggestion
            PermissionedIdentity::try_mutate(&identity, |pref| {
                pref.as_mut()
                    .ok_or_else(|| Error::<T>::NotExists.into())
                    .map(|p| p.intended_count += additional)
            });
```
(or something like that...)